### PR TITLE
[#9426] Show warning modal to alert users trying to permanently delete courses

### LINKED ISSUE #9426: Fix usage of traditional HTML alert in Instructor courses page
- **Environment**: * Local server at `V7.0.0`, `master` branch at commit `6078d96`

**Steps to reproduce**
1. Go to instructor courses page (`/web/instructor/courses`)
2. Delete a course
3. Delete that course permanently

**Expected behaviour**
Bootstrap alert is used (pointed by arrow in the screenshot below)

**Actual behaviour**
image
Traditional HTML alert is used instead

**Additional info**

Related to #9411

### PR DESCRIPTION
**Pull request by monmanuela**
Fixes #9426 

**Outline of Solution**
I added new components for the warning modal.
Now, when the user tries to permanently delete a course, this warning modal pops up:
image

Modal for permanently deleting all courses:
image

Modal for soft deletion of a course:
image

### CODE REVIEW

File: src/web/app/pages-instructor/instructor-courses-page/course-permanent-deletion-confirm-modal/course-permanent-deletion-confirm-modal.component.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  <h5 class="modal-title"><i class="fas fa-times-circle"></i> Confirm deleting a course</h5>
+  <button type="button" class="close" (click)="activeModal.dismiss()">
+    <i class="fas fa-times"></i>
+  </button>
+</div>
+<div class="modal-body">
+  <strong>Are you sure you want to permanently delete {{courseId}}?</strong>
+  <br>
+  This operation will delete all students and sessions in these courses.
+  All instructors of these courses will not be able to access them hereafter as well.
```
Comment (joanneong) [inline src/web/app/pages-instructor/instructor-courses-page/course-permanent-deletion-confirm-modal/course-permanent-deletion-confirm-modal.component.html]: For this section, all instances of "these courses" should just be "this course".

**Comment by monmanuela** [comment]
@joanneong Noted! Actually, I think I'll refactor one of the modals to warn users about soft deletion of a course. It's because the feedback session soft deletion has such modal, and I think it'll be good to unify the user flow. Does this sound good to you?

**Comment by monmanuela** [comment]
@joanneong Ready for review!

**Comment by monmanuela** [comment]
@wkurniawan07 Ready for review

**Pull request merged**